### ISSUE TITLE
x264 encoder failing to initialize

### ISSUE THREAD

**Issue by PMohanJ** [selkies-project/selkies#124]
The default software encoder x264 is failing to initialize while trying to access the desktop from browsers like brave and edge, saying the `height not divisible by 2`. 

Here are some logs:
![highlighted_logs](https://github.com/selkies-project/selkies-gstreamer/assets/57227290/363df120-1e50-4575-bb27-0b24ab57f89a)

Brave browser's resolution:  `1920x937`
![brave_browser](https://github.com/selkies-project/selkies-gstreamer/assets/57227290/6620680a-71e7-44d0-a283-45eed4a28d77)

Edge browser's resolution: `1860x927`
![edge](https://github.com/selkies-project/selkies-gstreamer/assets/57227290/e91a8d41-3f56-4483-ba2d-38a69e6f2229)

But in chrome it's working fine as the pixels of height is even numbered: `1920x912`
![chrome](https://github.com/selkies-project/selkies-gstreamer/assets/57227290/061eb8cd-afe8-4a87-9c8f-33c2b6f3c905)

Client device details: 
- windows 11
- Brave Version 1.61.104 Chromium: 120.0.6099.115 (Official Build) (64-bit)]
- Microsoft Edge Version 120.0.2210.77 (Official build) (64-bit)
- Chrome Version 120.0.6099.110 (Official Build) (64-bit)

Image used is:  ghcr.io/selkies-project/selkies-gstreamer/gst-py-example:main-ubuntu20.04

**Comment by ehfd** [comment]
Yeah, that is a known restriction in x264... The solution for now is to use an even viewport... I'll think of a solution.

**Comment by PMohanJ** [comment]
Hi @ehfd,

After looking into web-interface code part, I came across  below logic where the height is being calculated. 
```js
parseInt((document.body.offsetHeight - document.body.offsetHeight%2) * window.devicePixelRatio)
```

It seems like the point of subtracting the `offsetHeight%2` from `offsetHeight` is just to eliminate the odd numbered pixel. 
If that's the case, the reason for me still ending up with odd number is..
The DevicePixelRation of my device is `1.25`
brave browser offsetHeight=750   => (750 - 0) * 1.25 = 937.5

So if we just do that subtraction after the multiplication then problem solved. I've tried this and I no longer face that `not divisible by 2` error. 

Let me know what you think, I've got the changes ready for PR.
